src/compile/data/data.ts

```diff
@@ -58,17 +59,20 @@ export interface DataComponent {
   /** String set of time units that need their own data sources for scale domain */
   timeUnitDomain: StringSet;
 
-  /** Array of summary component object for producing summary (aggregate) data source */
-  summary: SummaryComponent[];
+  /** Array of summary component object for producing aggregates */
+  aggregate: AggregateComponent;
```

Comment:
Now I remember why I make my old `summary` an array of `SummaryComponent`s.

One major problem of "never" merging aggregate up like I did before is that now we have to recursively apply dimensions facet to more than just the child of current data. 

For example, if we have nesting like this.

```
facet
   layer
      unitA (aggregate - group by a)
      unitB (aggregate - group by b)
```

layer will not `aggregate` of `unitA` and `unitB`.  When you want to apply dimensions from the facet to `unitA` and `unitB`, you have to recursively apply it to deeper than one level.  (That's why original I always merge up the aggregate.) 

That said, we can keep your model if we find recursive application of facet not too terrible.  I guess you find my old array model is causing problem in other aspects too.  (In what aspects though?)

We should discuss about this in detail.